Docs: Edit of "ISSUE_TRIAGE" and other topics (part 3 of doc quality project) (#88106)
grafana/grafana: contribute/UPGRADING_DEPENDENCIES.md

# Guide to upgrading dependencies

> Upgrading Go or Node.js requires making changes in many different files. Refer to the list below for more information.

Reviewer: One more I caught at a scan

https://developers.google.com/style/word-list#below

```suggestion
Upgrading Go or Node.js requires making changes in many different files. Refer to the following list for more information.
```

go mod tidy
```

You have to commit the changes to `go.mod` and `go.sum` before you submit the pull request.

To understand what the actual dependencies of `grafana-server` are, you can run it with the `-vv` flag. Note that this command might produce an output different from `go.mod` contents, and `-vv` option is the source of truth here. The output lists the modules _compiled_ into the executable, whereas `go.mod` lists also test and weak transitive dependencies (that is, modules, used in some packages, which aren't in use by itself). If you're interested in reporting a vulnerability in a dependency module, consult the `-vv` output, maybe the "dependency" isn't actually a dependency as such.

### Upgrading dependencies

> If you need to upgrade a direct or indirect dependency, you can do it like so, `$MODULE` being the dependency in question: `go get -u $MODULE`. The corresponding entry in `go.mod` should then have the version you specified. If it's an indirect dependency, the entry should have the `// indirect` comment. 

Reviewer: ```suggestion
If you need to upgrade a direct or indirect dependency, you can do it like so, where _`MODULE`_ is the dependency in question: `go get -u <MODULE>`. The corresponding entry in `go.mod` should then have the version you specified. If it's an indirect dependency, the entry should have the `// indirect` comment. 
```

> - nodejs
> - golang

Reviewer: ```suggestion
- Node.js
- Go
```

- `grafana/build-container` (our custom Docker build container)

### grafana/build-container

The main build steps (in Drone) happen using a custom Docker image that comes pre-baked with some of the necessary dependencies.

Link: [`grafana/build-container`](https://github.com/grafana/grafana/tree/main/scripts/build/ci-build)

#### Dependencies

- fpm

> - nodejs
> - golang

Reviewer: ```suggestion
- Node.js
- Go
```
